### PR TITLE
[ADD] CheckboxList component, rework Checkbox component slightly

### PR DESCRIPTION
**Pull request by RHODOK**
User-story: 12652

Signed-off-by: Peter Alabaster <peter.alabaster@unipart.io>

### CODE REVIEW

File: src/core/components.py [hunk 1]
```diff
@@ -382,31 +382,83 @@ def get_validators(self):
...
-        "target",
     ]
 
     title = Translatable()
 
-    def __init__(self, title, data, target, **kwargs):
+    def __init__(self, title, destination_path, data, **kwargs):
         super().__init__(**kwargs)
         self.title = title
+        self.destination_path = destination_path
```
Comment (nocturnalastro) [inline src/core/components.py:498]: Not required

Comment (RHODOK) [reply src/core/components.py:498]: (y)

**Review by PeterAlabaster — APPROVED** [review]

File: src/core/schema/common.json [hunk 2]
```diff
@@ -102,6 +102,23 @@
...
       }
+    },
+    "checkboxlist_data": {
+      "type": "object",
+      "properties": {
+        "label": {
+          "type": "string"
+        },
+        "value": {
+          "oneOf": [
```
Comment (nocturnalastro) [inline src/core/schema/common.json:113]: Let's keep the JSON as explicit as possible and use a `*_path` keyword of the value is  a jsonpath you have. It makes it easier for the front end to know if it should be doing a look-up or not.

File: src/core/schema/components/checkbox.json [hunk 3]
```diff
@@ -6,10 +6,17 @@
...
-    "data": { "$ref": "../common.json#/definitions/jsonpath" },
-    "target": { "$ref": "../common.json#/definitions/jsonpath" }
+    "label": { "type": "string" },
+    "destination_path": { "$ref": "../common.json#/definitions/jsonpath" },
+    "value":
+    {
+      "oneOf": [
+        { "type": "string" },
+        { "type": "boolean" },
+        { "$ref": "../common.json#/definitions/jsonpath" }
```
Comment (nocturnalastro) [inline src/core/schema/components/checkbox.json]: same here also

File: src/core/schema/components/checkboxlist.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "type": "object",
+    "properties": {
+      "type": { "type": "string", "enum": ["checkboxlist"] },
+      "title": { "type": "string" },
+      "destination_path": { "$ref": "../common.json#/definitions/jsonpath" },
+      "data":
+      {
+        "oneOf": [
+          { "$ref": "../common.json#/definitions/checkboxlist_data" },
+          { "$ref": "../common.json#/definitions/jsonpath" }
```
Comment (nocturnalastro) [inline src/core/schema/components/checkboxlist.json]: and here

File: src/core/schema/common.json (hunk 2 above)
Comment (nocturnalastro) [reply src/core/schema/common.json:113]: If you want to you could have the class take either and then come up with some way of distinguishing between it being a jsonpath or anything else and output the correct field.

Comment (nocturnalastro) [reply src/core/schema/common.json:113]: @PeterAlabaster: I can't resolve this because of the permissions feel free to do it :)

Comment (PeterAlabaster) [reply src/core/schema/common.json:113]: Thanks man :)

**Review by ws-mps — APPROVED** [review]

**Review by joe-gb — APPROVED** [review]

**Pull request merged**